### ISSUE TITLE
RunSpecifiedTests feature for deployment

### ISSUE THREAD

**Issue by hpreets** [xjsender/haoide#96]
Currently I cannot find an option to run only selected tests while deployment. This feature would be very helpful while deploying to large orgs that take an hour or more to run the test classes.

**Comment by xjsender** [comment]
@hpreets , if you want to try this feature, you should upgrade your plugin to latest version 3.3.7.
Because this feature is not default enabled,  in order to let it work, you should change the value of ``testLevel`` in the ``deploy_options``setting to ``RunSpecifiedTests``, just like as below:
```json
// Deploy Options, 
// If you just want to valid a deployment,
// you can copy this setting to user settings and set checkOnly to true
// 
// Enhancement for issue #96
// testLevel: NoTestRun, RunSpecifiedTests, RunLocalTests, RunAllTestsInOrg
// If you want to run specified test classes when deploy, 
// you can set testLevel to RunSpecifiedTests
"deploy_options" : {
    "allowMissingFiles" : false,
    "autoUpdatePackage" : false,
    "checkOnly" : false,
    "ignoreWarnings" : true,
    "performRetrieve" : false,
    "purgeOnDelete" : false,
    "rollbackOnError" : true,
    "singlePackage" : true,
    "testLevel": "NoTestRun",
    "runTests" : ""
 },
```

**BTW, default setting is not changeable, you should put the change into your own user settings**

**Comment by hpreets** [comment]
Thanks @xjsender I upgraded to 3.3.7 package and added the above settings to user settings file and changed the setting variables but when I try to deploy it to production, it gives the error although I did specify "runTests"

sandboxname1: 2016.03.30 18:18:42: INVALID_OPERATION: runTests must not be empty when a testLevel of RunSpecifiedTests is used

![mwsnap 2016-03-30 18_25_12](https://cloud.githubusercontent.com/assets/3944578/14151271/0e89725e-f6a5-11e5-8718-5d596142e421.png)

Is there any settings I missed?

**Comment by xjsender** [comment]
@hpreets , you don't need to assign any value for ``runTests``, this feature is enabled for deploying whole package.

**Comment by xjsender** [comment]
@hpreets , when you execute ``deploy package`` command, plugin will let you choose test classes, and then deployment will start.

**Comment by hpreets** [comment]
Ah OK. yeah it worked that way. :-) Thanks @xjsender 

Normally how I deploy is right click on file >> HaoIDE >> Deploy To Server. The option to select Test classes was not coming this way.

**Comment by xjsender** [comment]
@hpreets , yes, I will add this feature in the next release.

**Comment by xjsender** [comment]
runSpecifiedTest support for file deploying is available since release 3.3.8

**Comment by Xtremefaith** [comment]
Can the "testLevel" not be set by "project", or at least overridden by a project's settings
